### DISCUSSION TITLE
Firecracker minimum RAM requirement

### DISCUSSION THREAD

**Discussion by zamazan4ik** [firecracker-microvm/firecracker#2733] [Q&A] [answered]
Hi!

I am trying to find any explanations, why 128 Mib of RAM was chosen as a minimal requirement for the VM? Why we cannot allocate a VM with e.g. 64 Mib of RAM?

I've tried to find any details about this design decision but without any success.

Thanks in advance!

**Comment by georgepisaltu** [comment, marked as answer]
Hi!

I answered on the issue you opened on the same topic (#2737), but I'll post this here for posterity.

I don't think it's a minimal requirement for VM memory in the sense that it's not enforced. I just confirmed this by booting up a VM with 64 MiB of RAM, then 48 and it still works. As long as your setup allows it, both host and guest configurations, you can go lower than 128 MiB of RAM.

I'd like to point out that Firecracker is oversubscribing memory by default using demand paging. This means that even if you start a 2 GiB VM, the actual memory used by the process will be much lower (probably around 150 MiB) until you "touch" that memory and is gets physically allocated on the host. I understand this does not address the need to put an upper limit to how much memory the guest is allowed to use, but it's worth mentioning.

**Reply by zamazan4ik** [reply]
Thanks!
